tariff/helper.go

```diff
@@ -12,3 +13,36 @@ func newBackoff() backoff.BackOff {
 	bo.MaxElapsedTime = time.Minute
 	return bo
 }
+
+type LeveledRate struct {
+	api.Rate
+	Cheap bool `json:"cheap"`
+}
+
+func LevelRates(rates api.Rates, percent float64) []LeveledRate {
+	res := make([]LeveledRate, len(rates))
+
+	var (
+		totalDuration time.Duration
+		weighedCost   float64
+	)
+
+	for _, r := range rates {
+		d := r.End.Sub(r.Start)
+		totalDuration += d
+		weighedCost += r.Price * float64(d)
+	}
+
+	if totalDuration > 0 {
+		weighedCost /= float64(totalDuration) * percent
+	}
+
+	for i, r := range rates {
+		res[i].Rate = r
+		if r.Price <= weighedCost {
+			res[i].Cheap = true
+		}
+	}
+
+	return res
+}
```

Comment:
Ich hätte erwartet, dass wir zusätzlich / mit den Rates einfach die ausgerechneten `avgCost` publishen. MMn. muss das nicht schon vorab in die einzelnen Rates reingebaut werden. In der UI würde ich vtml sowieso abhängig vom %-Wert noch den ct/g Betrag errechnen. Gleiches dann für das Highlighting im Tariff Chart.

Reply:
Siehe Vorschlag `smartCostRelativeBase` 👇